src/datagrid/DataGrid.tsx

```diff
@@ -294,13 +297,26 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
     // Function to return all selected rows
     getSelectedRows = (): DataGridRow[] => {
         const {allRows} = this.state;
-        let selectedRows = new Array();
-        if (this.state.selectAll) {
-            selectedRows = allRows;
-        } else {
-            selectedRows = allRows.filter(row => row["isSelected"] === true);
+        return allRows.filter(row => row.isSelected === true);
+    };
+
+    // Function to return all selection enabled rows
+    getSelectionEnabledRows = (allRows: DataGridRow[]): DataGridRow[] => {
+        return allRows.filter(row => !row.disableRowSelection);
```

Comment:
Why this method needs input,  why can't we use the this.state.allRows instead ?

Reply:
@maheshkulkarni4 I need this method before updating state. Hence passing rows as params.